### ISSUE TITLE
Using filter to modify outgoing subscription message affects other sockets

### ISSUE THREAD

**Issue by valkko** [hapijs/nes#212]
I'm trying to use the filter feature with subscriptions to partially filter out some data from the messages outgoing to specific clients.

The clients themselves can request to only receive specific part of the data, and this information I'm storing under the `socket.app` namespace. Then in the filter for that subscription, if that property in socket.app is set, I clone the message object, and remove any data the client doesn't want.

Now, if I understood the documentation correctly,  normally this would affect all the other sockets as well, which is why I clone the message first, before editing and returning it under `override` from the filter. For the clients without any filtering, I'm returning `true` from the filter.

Even then, the filter function randomly receives the already filtered message for rest of the sockets. Feels like this is based on the order the sockets have been opened, so if you do the filtering in the first opened socket, the filters for the other sockets after that receive the already filtered message.

In short, cloning the message object doesn't seem to actually prevent the filter changes from applying to all sockets.

**Comment by mtharrison** [comment]
Are you sure you're deep cloning the message rather than shallow cloning? Can you share an example?

**Comment by valkko** [comment]
Yes, I got it to reproduce with a flat object and object assign.

```
const dataFilter = (path, message, options) => {
  console.log('Filter started with', message) // at this point, the message should have both a and b properties
  if (options.socket.app.filter) { // filter should be 'a' or 'b'
    // Cloning the message, so it shouldn't affect the other sockets
    const newMessage = Object.assign({}, message)
    delete newMessage[options.socket.app.filter]
    return { override: newMessage }
  }
  return true
}
```

Basically, I'm pushing objects like this through it:
```
{ a: 1, b: 1 }
```

And the filter `app.filter` here is either 'a' or 'b'.

**Comment by mtharrison** [comment]
@valkko please try out 7.0.3 when you have chance.
